tls/s2n_tls13_secrets.h

```diff
@@ -41,6 +40,8 @@ struct s2n_tls13_secrets {
     uint8_t client_app_secret[S2N_TLS13_SECRET_MAX_LEN];
     uint8_t server_app_secret[S2N_TLS13_SECRET_MAX_LEN];
     uint8_t resumption_master_secret[S2N_TLS13_SECRET_MAX_LEN];
+
+    s2n_extract_secret_type_t extract_secret_type;
 };
```

Comment:
This is a bit of a hack.

The TLS1.3 secrets and the TLS1.2 secrets are fields in a union so share memory. Setting the TLS1.2 master secret when resuming a session overwrote the extract_secret_type field because of its early position in the structure. When we fell back to a full TLS1.3, that was a problem because extract_secret_type wasn't 0 as expected.

I could reset extract_secret_type in this case, but it seems safer to move this field out of the union, since it can be used as a "type" field to determine whether we're using TLS1.2 or TLS1.3. There's not memory cost to doing so; the TLS1.3 part of the union is much bigger than the TLS1.2 part.

Reply:
Because if extract_secret_type is NONE like it should be, we don't attempt to read any secrets, just write them. "extract_secret_type" basically defines which fields in the struct are set.